src/main/CommandHandler.cpp

```diff
@@ -75,7 +75,8 @@ CommandHandler::CommandHandler(Application& app) : mApp(app)
 
     mServer->add404(std::bind(&CommandHandler::fileNotFound, this, _1, _2));
 
-    if (mApp.getConfig().MODE_STORES_HISTORY)
+    if (mApp.getConfig().MODE_STORES_HISTORY &&
+        mApp.getConfig().MODE_STORES_HISTORY_LEDGERHEADERS)
```

Comment:
this should be an `||` as `maintenance` related endpoints should work regardless of which historical data is in there